### PR TITLE
EI 203 apply amend apprenticeships selection

### PR DESCRIPTION
**Pull request by ruslanss**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives.Data/IncentiveApplication/IncentiveApplicationDataRepository.cs [hunk 1]
```diff
@@ -39,8 +43,22 @@ public async Task Update(IncentiveApplicationModel incentiveApplication)
...
             {
                 _dbContext.Entry(existingApplication).CurrentValues.SetValues(model);
+                _dbContext.RemoveRange(existingApplication.Apprenticeships);
+                _dbContext.AddRange(model.Apprenticeships);
+
                 await _dbContext.SaveChangesAsync();
             }
         }
+
+        private static IEnumerable<IncentiveApplicationApprenticeship> RemovedApprenticeships(Models.IncentiveApplication application, Models.IncentiveApplication model)
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Data/IncentiveApplication/IncentiveApplicationDataRepository.cs:53]: Are these still used?

Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Data/IncentiveApplication/IncentiveApplicationDataRepository.cs:53]: nope, good spot!

File: src/SFA.DAS.EmployerIncentives.Api/Types/UpdateIncentiveApplicationRequest.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using SFA.DAS.EmployerIncentives.Abstractions.DTOs.Commands;
+using System;
+using System.Collections.Generic;
+
+namespace SFA.DAS.EmployerIncentives.Api.Types
+{
+    public class UpdateIncentiveApplicationRequest
+    {
+        public Guid IncentiveApplicationId { get; set; }
+        public long AccountId { get; set; }
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Api/Types/UpdateIncentiveApplicationRequest.cs:10]: Do we need accountid and accountlegalentityId on the request?

Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Api/Types/UpdateIncentiveApplicationRequest.cs:10]: well, in theory we don't need either of them. added for consistency / future checks?

Comment (Stephek) [reply src/SFA.DAS.EmployerIncentives.Api/Types/UpdateIncentiveApplicationRequest.cs:10]: YAGNI !

Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Api/Types/UpdateIncentiveApplicationRequest.cs:10]: YAGNI it is then 😄 -- REMOVED ❌

**Pull request merged**